src/views/Alteration.vue

```diff
@@ -142,6 +157,25 @@ export default class Alteration extends Mixins(LegalApiMixin, FilingTemplateMixi
     Vue.nextTick(() => this.setHaveChanges(false))
   }
 
+  /** Fetch Business Snapshot */
+  private async fetchBusinessSnapshot (): Promise<any> {
+    const businessData = await this.getBusinessData()
+    const businessAliases = await this.getBusinessData(BusinessDataTypes.TRANSLATIONS)
+    const businessAddresses = await this.getBusinessData(BusinessDataTypes.ADDRESSES)
+    const businessDirectors = await this.getBusinessData(BusinessDataTypes.DIRECTORS)
+    const businessShareStructure = await this.getBusinessData(BusinessDataTypes.SHARE_CLASSSES)
+    const contactInfo = await this.getContactInfo()
```

Comment:
You should do these in parallel!

https://github.com/bcgov/business-filings-ui/blob/acddf879c61ecda163ced5e8b236d2768ffb08f6/src/App.vue#L311

Reply:
oh i like it! Will do, thats awesome.

Reply:
Fixed in next commit. Really like that approach!

Reply:
FYI, the browser can handle several simultaneous/pending network calls. I don't know what the current max is but I think it's over 10; anything over that has to wait for a previous call to complete.

As you can surmise, this shortens the fetch time from "sum of all fetches" to "time of longest fetch". This also allows increased performance/optimization of the web server/API through caching or multiple instances (ie, scaling).

Reply:
Yea, this was an excellent way to do it. It actually improved the laod time significantly 👍 Thanks!